src/services/tipStatus.ts

```diff
@@ -0,0 +1,100 @@
+import config from "config";
+
+import { Pool } from "pg";
+import { Request, Response } from "express";
+
+const safeBlockDifference = parseInt(config.get("safeBlockDifference"));
+
+const bestBlockQuery = `
+  SELECT epoch_no AS "epoch",
+    epoch_slot_no AS "slot",
+    encode(hash, 'hex') as hash,
+    block_no AS height
+  FROM BLOCK
+  ORDER BY id DESC
+  LIMIT 1;`;
+
+const safeBlockQuery = `SELECT epoch_no AS "epoch",
+  epoch_slot_no AS "slot",
+  encode(hash, 'hex') as hash,
+  block_no AS height
+FROM BLOCK
+WHERE block_no <= (SELECT MAX(block_no) FROM block) - ($1)::int
+ORDER BY id DESC
+LIMIT 1;
+`;
+
+const bestBlockFromReferenceQuery = `SELECT encode(hash, 'hex') as "hash", block_no as "blockNumber"
+    FROM block
+    WHERE encode(hash, 'hex') = any(($1)::varchar array)
+      AND block_no IS NOT NULL
+    ORDER BY block_no DESC
+    LIMIT 1`;
+
+const safeBlockFromReferenceQuery = `SELECT encode(hash, 'hex') as "hash", block_no as "blockNumber"
+  FROM block
+  WHERE encode(hash, 'hex') = any(($1)::varchar array)
+    AND block_no IS NOT NULL
+    AND block_no <= (SELECT MAX(block_no) FROM block) - ($2)::int
+  ORDER BY block_no DESC
+  LIMIT 1`;
+
+const getBestAndSafeBlocks = (pool: Pool) => async (): Promise<{
+  safeBlock: string | undefined,
+  bestBlock: string | undefined
+}> => {
+  const bestBlockResult = await pool.query(bestBlockQuery);
+  const safeBlockResult = await pool.query(safeBlockQuery, [safeBlockDifference]);
+
+  return {
+    safeBlock: safeBlockResult.rowCount > 0 ? safeBlockResult.rows[0].hash : undefined,
+    bestBlock: bestBlockResult.rowCount > 0 ? bestBlockResult.rows[0].hash : undefined,
+  };
+};
+
+export const handleTipStatusGet = (pool: Pool) => async (req: Request, res: Response) => {
+  const result = await getBestAndSafeBlocks(pool)();
```

Comment:
Not clear why `getBestAndSafeBlocks` itself has to return a function if every single time it's being used right there right away with this double-call `foo(pool)()` 🤔

Reply:
That was a bad copy paste 😅 
Fixed!

Reply:
😄 👌